pandas/io/parsers.py

```diff
@@ -3420,11 +3428,11 @@ def get_rows(self, n, skiprows=None):
         self.buffer = iter(buffer_rows)
         return detect_rows
 
-    def detect_colspecs(self, n=100, skiprows=None):
+    def detect_colspecs(self, infer_nrows=100, skiprows=None):
```

Comment:
Same as above

Reply:
Oh, I actually see another problem here. I co-opted `n` because it was performing the intended function of `infer_nrows`, but that would also break previous code that relied on calling `n` explicitly. I'll return the code and tack on `infer_nrows` at the end, like you pointed out.